targets/TARGET_Cypress/TARGET_PSOC6/mbed_rtx.h

```diff
@@ -21,6 +21,10 @@
 #ifndef MBED_MBED_RTX_H
 #define MBED_MBED_RTX_H
 
-/* This file is required, but doesn't actually need to do anything. */
+/*
+ * This is used to obtain Object specific Memory allocation required for
+ * mutex creation in an efficient way.
+ */
+#define OS_MUTEX_OBJ_MEM    1
```

Comment:
Sorry guys, this is not supported memory model in Mbed OS. Please use statically allocated memory for your mutexes.